b3lb/rest/endpoints/b3lb/endpoints.py

```diff
@@ -191,26 +191,39 @@ async def create(request, endpoint, params, node, secret):
 
     # check if records are enabled
     if secret.is_record_enabled:
-        record_relation = RecordSet()
-        record_relation.secret = secret
-        record_relation.meeting_id = meeting_id
-        record_relation.record_available_url = "https:://{}-{}.{}/{}".format(secret.tenant.slug.lower(), str(secret.sub_id).zfill(3), settings.B3LB_API_BASE_DOMAIN, "b3lb/b/record/available")
-
+        record_ready_url = "https://{}-{}.{}/{}".format(secret.tenant.slug.lower(), str(secret.sub_id).zfill(3), settings.B3LB_API_BASE_DOMAIN, "b3lb/b/record/available")
+        if "meta_meta_bbb-recording-ready-url" in params:
+            record_set = await sync_to_async(RecordSet.objects.create)(secret=secret, meeting_id=meeting_id, record_ready_origin_url=params["meta_meta_bbb-recording-ready-url"])
+        else:
+            record_set = await sync_to_async(RecordSet.objects.create)(secret=secret, meeting_id=meeting_id)
+        params["meta_meta_bbb-recording-ready-url"] = "{}?nonce={}".format(record_ready_url, record_set.nonce)
     else:
-        # suppress any record related parameter
+        # record aren't enabled -> suppress any record related parameter
+        record_set = None
         for param in [Parameter.RECORD, Parameter.ALLOW_START_STOP_RECORDING, Parameter.AUTO_START_RECORDING]:
             params[param] = "false"
 
-    response = await pass_through(request, endpoint, params, node, body=body)
+    if "meta_endCallbackUrl" in params:
+        end_callback_origin_url = params["meta_endCallbackUrl"]
+    else:
+        end_callback_origin_url = ""
 
     defaults = {
         "id": meeting_id,
         "secret": secret,
         "node": node,
-        "room_name": params.get("name", "Unknown")
+        "room_name": params.get("name", "Unknown"),
+        "end_callback_url": end_callback_origin_url
     }
 
-    obj, created = await sync_to_async(Meeting.objects.get_or_create)(id=meeting_id, secret=secret, defaults=defaults)
+    meeting, created = await sync_to_async(Meeting.objects.get_or_create)(id=meeting_id, secret=secret, defaults=defaults)
+
+    if record_set:
+        params["meta_endCallbackUrl"] = "https://{}-{}.{}/{}?nonce={}&end_nonce={}".format(secret.tenant.slug.lower(), str(secret.sub_id).zfill(3), settings.B3LB_API_BASE_DOMAIN, "b3lb/b/meeting/end",  record_set.nonce, meeting.end_nonce)
```

Comment:
I don't think `end_nonce` is required. The recording-is-available should IMHO not be passed to the BBB nodes but recorded in b3lb. The post recording hook needs to upload der record using the meeting id, the recording-is-available would be triggered by the post processing celery task.

Reply:
record ready will now be dropped from create request and stored in recordSet.